src/gsCore/gsDofMapper.h

```diff
@@ -346,9 +366,23 @@ class GISMO_EXPORT gsDofMapper
 
     /// Returns true if \a gl is a coupled dof.
     inline bool is_coupled_index( index_t gl) const 
-    {           
-        return  (gl < m_numFreeDofs + m_shift                    ) && // is a free dof and
-                (gl + m_numCpldDofs + 1 > m_numFreeDofs + m_shift);   // is not standard dof
+    {
+	    return  (gl < m_numFreeDofs + m_shift                    ) && // is a free dof and
+                    (gl + m_numCpldDofs + 1 > m_numFreeDofs + m_shift);   // is not standard dof
+    }
+
+    /// Returns true if \a gl is a tagged dof.
+    inline bool is_tagged( index_t i, index_t k = 0 ) const
+    { return  is_tagged_index( index(i, k) ); }
+
+    /// Returns true if \a gl is a tagged dof.
+    inline bool is_tagged_index( index_t gl) const
+    {
+        if(m_tagged.empty())
+            return  (gl < m_numFreeDofs + m_shift                    ) && // is a free dof and
+                    (gl + m_numCpldDofs + 1 > m_numFreeDofs + m_shift);   // is not standard dof
+        else
+            return std::binary_search(m_tagged.begin(),m_tagged.end(),gl);
```

Comment:
Should be just "return std::binary_search(m_tagged.begin(),m_tagged.end(),gl);"

Reply:
Ups, you are right